### PR TITLE
add adjustment on secure compose button for new Gmail UI

### PR DESCRIPTION
**Pull request by martgil**
This PR will adjust the secure compose button UI for the new Gmail UI.

proposed changes looks like:
Screen Shot 2022-03-07 at 3 16 50 PM

close https://github.com/FlowCrypt/flowcrypt-browser/issues/4308

issue #0000 // if it doesn't close the issue yet

----------------------------------

**Tests** _(delete all except exactly one)_:
- Does not need tests (refactor only, docs or internal changes)

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Comment by martgil** [comment]
Hello sir, @tomholub I would like to propose a patch to the new Gmail UI issue (we've been receiving a few concerns about it lately). please let me know if it looks good to you. thank you!

File: extension/js/common/xss-safe-factory.ts [hunk 1]
```diff
@@ -233,8 +233,12 @@ export class XssSafeFactory {
...
       return `<div class="_fce_c ${this.destroyableCls} cryptup_compose_button_container" role="presentation">${btn}</div>`;
     } else {
-      const btn = `<div class="new_secure_compose_window_button" id="flowcrypt_secure_compose_button" role="button" tabindex="0" data-test="action-secure-compose">Secure Compose</div>`;
-      return `<div class="${this.destroyableCls} z0">${btn}</div>`;
+      let btn = `<div class="new_secure_compose_window_button" id="flowcrypt_secure_compose_button" role="button" tabindex="0" data-test="action-secure-compose">Secure Compose</div>`;
+      if ($('.V6.CL.W9').length === 1) { // if mail primary button present for new Gmail UI
+        btn = `<div class="new_secure_compose_window_button compose_button_simple" id="flowcrypt_secure_compose_button" role="button" tabindex="0" data-test="action-secure-compose">
+        </div><div class="apW">FlowCrypt</div>`;
+      }
+      return `<div class="${this.destroyableCls} z0 ${$('.V6.CL.W9').length === 1 ? 'pb-10px' : '' }">${btn}</div>`;
```
Comment (rrrooommmaaa) [inline extension/js/common/xss-safe-factory.ts:241]: I'd suggest to create a `const isNewGmailUI = $('.V6.CL.W9').length === 1;`
and use it in both places
so we don't end up fixing the first case and forgetting the other one if something changes.

Comment (tomholub) [reply extension/js/common/xss-safe-factory.ts:241]: Agree, let's call it `isNewGmailUi2022`

**Comment by tomholub** [comment]
@martgil this looks good. With the small update above, this would be good for merging. Thanks for looking into this! I had a hard time trying to reproduce.

File: extension/js/common/xss-safe-factory.ts (hunk 1 above)
Comment (martgil) [reply extension/js/common/xss-safe-factory.ts]: Thank you, sir, @rrrooommmaaa, and sir @tomholub for the review. 

I've also added a few css tweaks to the UI to maintain it's signature "secure compose" look while adapting to the new ui:
Screen Shot 2022-03-08 at 9 00 57 AM

**Comment by tomholub** [comment]
@martgil please check if hovering above the new button will produce a "Secure Compose" title

**Review by tomholub — APPROVED** [review]

**Pull request merged**